airflow/contrib/hooks/winrm_hook.py

```diff
@@ -17,121 +17,191 @@
 # specific language governing permissions and limitations
 # under the License.
 #
+
 import getpass
+
 from winrm.protocol import Protocol
+
 from airflow.exceptions import AirflowException
 from airflow.hooks.base_hook import BaseHook
 from airflow.utils.log.logging_mixin import LoggingMixin
 
 
 class WinRMHook(BaseHook, LoggingMixin):
-
     """
     Hook for winrm remote execution using pywinrm.
 
-    :param ssh_conn_id: connection id from airflow Connections from where all
-        the required parameters can be fetched like username, password or key_file.
+    :seealso: Refer to https://github.com/diyan/pywinrm/blob/master/winrm/protocol.py
+        for parameter values.
+
+    :param str ssh_conn_id: connection id from airflow Connections from where all
+        the required parameters can be fetched like username and password.
         Thought the priority is given to the param passed during init
-    :type ssh_conn_id: str
```

Comment:
Why have you removed the **types** for each parameter?

Reply:
I moved them inline (see http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists). I can put them back if you don't like that style.

Reply:
To be consistent with the other modules, I think it would be better to keep it as it was.

Reply:
Fixed